pkg/api/goprobe/server/config.go

```diff
@@ -2,10 +2,80 @@ package server
 
 import (
 	"net/http"
+	"net/url"
+	"strings"
 
+	"github.com/els0r/goProbe/cmd/goProbe/config"
+	gpapi "github.com/els0r/goProbe/pkg/api/goprobe"
 	"github.com/gin-gonic/gin"
 )
 
-func (server *Server) postConfig(c *gin.Context) {
-	c.JSON(http.StatusNotImplemented, nil)
+func (server *Server) getConfig(c *gin.Context) {
+	iface := c.Param(ifaceKey)
+	ifaces := c.Request.URL.Query().Get(gpapi.IfacesQueryParam)
+
+	resp := &gpapi.ConfigResponse{}
+	resp.StatusCode = http.StatusOK
+
+	var err error
+	ifaces, err = url.QueryUnescape(ifaces)
+	if err != nil {
+		resp.StatusCode = http.StatusBadRequest
+		resp.Error = err.Error()
+
+		c.AbortWithStatusJSON(resp.StatusCode, resp)
+		return
+	}
+
+	ctx := c.Request.Context()
+
+	if iface != "" {
+		resp.Ifaces = server.captureManager.Config(ctx, iface)
+	} else {
+		if ifaces != "" {
+			// fetch all specified
+			resp.Ifaces = server.captureManager.Config(ctx, strings.Split(ifaces, ",")...)
+		} else {
+			// otherwise, fetch all
+			resp.Ifaces = server.captureManager.Config(ctx)
+		}
+	}
+
+	if len(resp.Ifaces) == 0 {
+		resp.StatusCode = http.StatusNoContent
```

Comment:
IIRC: Technically, if a `HTTP No Content` is returned the content / body should be empty. Sending the response as body at the same time "violates" that I guess...

Reply:
Does `httpc` handle a `nil` Body well? If so, then I can just return an `EmptyResults` error in the client.

Reply:
I don't think over the wire you can send an actual `nil` body. If the server sets an empty (or nil, which is probably the default anyway) body then HTTP will just have, well, no body. So if you return an `HTTP No Content` and then handle that it shouldn't matter what the body is (worst case you can exclude  `HTTP No Content` from the list of acceptable return codes and actually receive an error from `httpc` (using `AcceptedResponseCodes()`).